### DISCUSSION TITLE
Default selected option for LookupChoiceFilter ?

### DISCUSSION THREAD

**Discussion by Kagee** [carltongibson/django-filter#1619] [Q&A] [answered]
I have read docs, source code, and more source code - is there no way to configure a default selected value for the LookupChoiceFilter  dropdown?

Django 3.2
django-filter 23.3

I currently have a LookupChoiceFilter as follows
````
name = django_filters.LookupChoiceFilter(
     lookup_choices=[
        ('icontains', 'Contains'),
        ('istartswith', 'Starts with'),
        ('iexact', 'Equals'),
    ],
    # empty_label='Contains', # Only changes the label from ----- to Contains
    # empty_value='icontains', # does nothing
    # lookup_expr=('icontains', 'Contains'), # does nothing
    # lookup_expr='icontains', # does nothing
    # required=True, # just makes both fields required, user must still select from dropdown
    # label="exact",  # does nothing     
    # null_value='icontains', # Field.__init__() got an unexpected keyword argument '...'
    # default=('icontains', 'Contains'),  # Field.__init__() got an unexpected keyword argument '...'
    # default='icontains', # Field.__init__() got an unexpected keyword argument '...'
    )
````

**Comment by Kagee** [comment]
Alternatively, is it possible for me to supply the whole dropdown form element myself?

**Comment by carltongibson** [comment]
You should be able to pass the `initial` kwarg, that will get passed to the form field. https://docs.djangoproject.com/en/4.2/ref/forms/fields/#django.forms.Field.initial

**Reply by Kagee** [reply]
Neither of these three worked...

````
class OrderItemFilter(django_filters.FilterSet):
      name = django_filters.LookupChoiceFilter(
           lookup_choices=[
          ('icontains', 'Contains'),
          ('istartswith', 'Starts with'),
          ('iexact', 'Equals'),
      ],
      # initial=('icontains', 'Contains'),  # does nothing
      # initial='icontains',  # does nothing
      # initial='Contains',  # does nothing
     )
````

**Comment by Kagee** [comment]
Just for fun i also tried this, no worky
````
class OrderItemFilter(django_filters.FilterSet):
      name = django_filters.LookupChoiceFilter(
           lookup_choices=[
          ('icontains', '---------'), # Just created a extra optionwith label --------- and value icontains
          ('icontains', 'Contains'),
          ('istartswith', 'Starts with'),
          ('iexact', 'Equals'),
      ],
      )
````

**Comment by carltongibson** [comment]
See the test case here: 

https://github.com/carltongibson/django-filter/blob/4a618e080ac904cb9bbcd0066878d3e06014437e/tests/test_forms.py#L161-L170

... but you have to pass the initial data in the format expect for the multi-value field... `[..., 'icontains']` I suppose.

**Comment by Kagee** [comment, marked as answer]
Solution: `empty_label=None,`

I could get no version of `inital` to work. 

After having read hundreds of documentation pages and source code lines for both  Django and django-filters i found that this will work - there will be no empty option element, and the first tuple in `lookup_choices` will be the top of the list. It will not have the `selected` HTML attribute, but still works.

If this is how it is supposed to be done i __strongly__ suggest it is added to the documentation, optionally the *correct* way.

````
class OrderItemFilter(django_filters.FilterSet):
      name = django_filters.LookupChoiceFilter(
           lookup_choices=(
              ('icontains', 'Contains'),
              ('istartswith', 'Starts with'),
              ('iexact', 'Equals'),
          ),
      empty_label=None,
      )

      class Meta:
          model = OrderItem
````
